pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -142,7 +144,9 @@ public SegmentGeneratorConfig(TableConfig tableConfig, Schema schema) {
     Preconditions.checkNotNull(schema);
     Preconditions.checkNotNull(tableConfig);
     _timestampIndexConfigs.putAll(extractTimestampIndexConfigsFromTableConfig(tableConfig));
-    setSchema(updateSchemaWithTimestampIndexes(schema, _timestampIndexConfigs));
+    Schema updatedSchema = updateSchemaWithTimestampIndexes(schema, _timestampIndexConfigs);
```

Comment:
This actually fails in IntermediateSegmentsTest. Reason being if `$INCOMPLETE_RECORD_KEY$` column value is null the indexing fails.
Either we need to ensure to set this column to `false` instead of leaving it as `null` OR need to think about another solution

Reply:
it's not picking up default value false, because this is added after the transformers?

Reply:
we could just add this upfront? for rt before the consumption begins and for offline during the SegmentCreationDriver init?

Reply:
Yeah, will do that.

Reply:
So based upon the discussions, I am going to remove this method and we are going to go with aggregate metrics.